src/OrchardCore.Modules/OrchardCore.Contents/Filters/BuildDisplayFilter.cs

```diff
@@ -13,9 +14,19 @@ public class BuildDisplayFilter : ILiquidFilter
     {
         public async Task<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, TemplateContext ctx)
         {
-            var contentItem = input.ToObjectValue() as ContentItem;
+            var obj = input.ToObjectValue();
 
-            if (contentItem == null)
+            if (!(obj is ContentItem contentItem))
+            {
+                contentItem = null;
+
+                if (obj is JObject jObject)
+                {
+                    contentItem = jObject.ToObject<ContentItem>();
+                }
+            }
+
+            if (contentItem == null || contentItem.ContentItemId == null)
```

Comment:
Why do you test `ContentItemId`?

Reply:
If the input is a `JObject` but which not represents a `ContentItem`, then a `ContentItem` is still created, but where properties are equal to their default value or null (unless `.Data and .Content` which get some values).

So, to prevent it from failling, we check one of these properties which is null in this case.